tests/fixture/wrangler_toml.rs

```diff
@@ -0,0 +1,100 @@
+use std::collections::HashMap;
+
+use serde::Serialize;
+
+// small suite of flexible toml structs
+// the  idea here is to focus on "when this config key is set"
+// rather than needing to write tomls all the time.
+// these structs set every value as an `Option`. To use,
+// initialize a new WranglerToml::default() and begin setting
+// values on it.
+#[derive(Clone, Debug, Default, Serialize)]
+pub struct KvConfig<'a> {
+    pub binding: Option<&'a str>,
+    pub id: Option<&'a str>,
+}
+
+#[derive(Clone, Debug, Default, Serialize)]
+pub struct SiteConfig<'a> {
+    pub bucket: Option<&'a str>,
+    #[serde(rename = "entry-point")]
+    pub entry_point: Option<&'a str>,
+    pub include: Option<Vec<&'a str>>,
+    pub exclude: Option<Vec<&'a str>>,
+}
+
+#[derive(Clone, Debug, Default, Serialize)]
+pub struct EnvConfig<'a> {
+    pub name: Option<&'a str>,
+    pub account_id: Option<&'a str>,
+    pub workers_dev: Option<bool>,
+    pub route: Option<&'a str>,
+    pub routes: Option<Vec<&'a str>>,
+    pub zone_id: Option<&'a str>,
+    pub webpack_config: Option<&'a str>,
+    pub private: Option<bool>,
+    pub site: Option<SiteConfig<'a>>,
+    #[serde(rename = "kv-namespaces")]
+    pub kv_namespaces: Option<Vec<KvConfig<'a>>>,
+}
+
+#[derive(Clone, Debug, Default, Serialize)]
+pub struct WranglerToml<'a> {
+    pub name: Option<&'a str>,
+    #[serde(rename = "type")]
+    pub target_type: Option<&'a str>,
+    pub account_id: Option<&'a str>,
+    pub workers_dev: Option<bool>,
+    pub route: Option<&'a str>,
+    pub routes: Option<Vec<&'a str>>,
+    pub zone_id: Option<&'a str>,
+    pub webpack_config: Option<&'a str>,
+    pub private: Option<bool>,
+    pub env: Option<HashMap<&'a str, EnvConfig<'a>>>,
+    #[serde(rename = "kv-namespaces")]
+    pub kv_namespaces: Option<Vec<KvConfig<'a>>>,
+    pub site: Option<SiteConfig<'a>>,
```

Comment:
the only problem i have with this pr is all of the lifetimes - it seems to me that the `<&'a str>` could all be `String` just for readability sake but I might be missing something here. I know we talked about this yesterday but wanted to leave this here regardless.

Reply:
i mean honestly, they're all concentrated here, and the only reason to do this is to simplify the calls whenever we add something (i.e. don't need to call to_string on everything). no need to allocate a whole String for each of these when a slice will do. i also think it wouldn't hurt for us to be comfortable with seeing explicit lifetimes in some of our code (again, explicit, because the lifetimes are always there, they are just implicit everywhere).

Reply:
i am really on board with the idea of trying to reduce `String` declarations for the sake of good memory practices :)